guides/source/sign_up_and_settings.md

```diff
@@ -0,0 +1,1659 @@
+  </div>
+
+  <div>
+    <%= form.submit "Sign up" %>
+  </div>
+<% end %>
+```
+
+This form collects the user's name, email, and password. We're using the `autocomplete` attribute to help the browser suggest the values for these fields based on the user's saved information.
+
+You'll also notice we set `url: sign_up_path` in the form alongside `model: @user`. Without this `url:` argument, `form_with` would see we have a `User` and send the form to `/users` by default. Since we want the form to submit to `/sign_up`, we set the `url:` to override the default route.
+
+Back in `app/controllers/sign_ups_controller.rb` we can handle the form submission by adding the `create` action.
+
+```ruby#6-19
+class SignUpsController < ApplicationController
+  def show
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(sign_up_params)
+    if @user.save
+      start_new_session_for(@user)
+      redirect_to root_path
+    else
+      render :show, status: :unprocessable_entity
+    end
+  end
+
+  private
+    def sign_up_params
+      params.expect(user: [ :first_name, :last_name, :email_address, :password, :password_confirmation ])
+    end
+end
+```
+
+The `create` action assigns parameters and attempts to save the user to the database. If successful, it logs the user in and redirects to `root_path`, otherwise it re-renders the form with errors.
+
+Authenticated users can still access `SignUpsController` and create another account while they're logged in which can be confusing.
+
+Let's fix this by adding a helper to the `Authentication` module in `app/controllers/concerns/authentication.rb`.
+
+```ruby#14-17
+module Authentication
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :require_authentication
+    helper_method :authenticated?
+  end
+
+  class_methods do
+    def allow_unauthenticated_access(**options)
+      skip_before_action :require_authentication, **options
+    end
+
+    def unauthenticated_access_only(**options)
+      allow_unauthenticated_access **options
+      before_action -> { redirect_to root_path if authenticated? }, **options
+    end
+
+    # ...
+```
+
+The `unauthenticated_access_only` class method can be used in any controller where we want to restrict actions to unauthenticated users only.
+
+We can then use this method at the top of `SignUpsController`.
+
+```ruby#2
+class SignUpsController < ApplicationController
+  unauthenticated_access_only
+
+  # ...
+end
+```
+
+### Rate Limiting Sign Up
+
+Our application will be accessible on the internet so we're bound to have malicious bots and users trying to spam our application. We can add rate limiting to sign up to slow down anyone submitting too many requests.
+
+Rails makes this easy with the [`rate_limit`](https://api.rubyonrails.org/classes/ActionController/RateLimiting/ClassMethods.html) method in controllers.
+
+```ruby#3
+class SignUpsController < ApplicationController
+  unauthenticated_only_access
```

Comment:
The callback is currently named `unauthenticated_only_access`, but the existing helpers follow the `*_access_only` pattern.  
Would you mind renaming it to `unauthenticated_access_only` for consistency?

```suggestion
  unauthenticated_access_only
```